pettingzoo/utils/wrappers/order_enforcing.py

```diff
@@ -134,13 +107,25 @@ def __iter__(self) -> AECOrderEnforcingIterator[AgentID, ObsType, ActionType]:
 
 
 class AECOrderEnforcingIterator(AECIterator[AgentID, ObsType, ActionType]):
-    def __next__(self) -> AgentID:
-        agent = super().__next__()
+    def __init__(
+        self, env: OrderEnforcingWrapper[AgentID, ObsType, ActionType], max_iter: int
+    ):
         assert hasattr(
-            self.env, "_has_updated"
+            env, "_has_updated"
         ), "env must be wrapped by OrderEnforcingWrapper"
+        # this is set during the super call to init, so setting it here
+        # is redundant. However, it silences pyright errors because it tells
+        # pyright that self.env is an OrderEnforcingWrapper (which may not be
+        # strictly true, but it should have OrderEnforcingWrapper somewhere
+        # in the wrapper list). This might be better handled by Protocols,
+        # but this approach works.
+        self.env = env  # silence pyright errors
```

Comment:
Imo we’re better off just ignoring the pyright error directly vs setting a variable too early or having extra variables that aren’t set exactly how they should be, simpler that way

Reply:
fair enough. I changed this back.